pkg/processor/build/builder.go

```diff
@@ -106,6 +112,16 @@ func (b *Builder) Build(options *platform.BuildOptions) (*platform.BuildResult,
 
 	b.logger.InfoWith("Building", "name", b.options.FunctionConfig.Meta.Name)
 
+	// create a staging directory
+	b.tempDir, err = b.createStagingDir()
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed create staging directory")
+	}
+
+	if !options.FunctionConfig.Spec.Build.NoCleanup {
```

Comment:
Might be cleaner here to `defer` the cleanup function call unconditionally and check `NoCleanup` from it instead. This will allow to add a log message along the lines of "--no-cleanup flag provided, skipping temporary dir cleanup"